Collector/ProfilePlugin.php

```diff
@@ -77,15 +77,60 @@ public function handleRequest(RequestInterface $request, callable $next, callabl
             return $first($request);
         };
 
-        return $this->plugin->handleRequest($request, $wrappedNext, $wrappedFirst)->then(function (ResponseInterface $response) use ($profile) {
+        try {
+            $promise = $this->plugin->handleRequest($request, $wrappedNext, $wrappedFirst);
+        } catch (Exception $e) {
+            $profile->setFailed(true);
+            $profile->setResponse($this->formatter->formatException($e));
+            $this->collectRequestInformation($request, $stack);
+
+            throw $e;
+        }
+
+        return $promise->then(function (ResponseInterface $response) use ($profile, $request, $stack) {
             $profile->setResponse($this->formatter->formatResponse($response));
+            $this->collectRequestInformation($request, $stack);
 
             return $response;
-        }, function (Exception $exception) use ($profile) {
+        }, function (Exception $exception) use ($profile, $request, $stack) {
             $profile->setFailed(true);
             $profile->setResponse($this->formatter->formatException($exception));
+            $this->collectRequestInformation($request, $stack);
 
             throw $exception;
         });
     }
+
+    /**
+     * Collect request information when not already done by the HTTP client. This happens when using the CachePlugin
+     * and the cache is hit without re-validation.
+     *
+     * @param RequestInterface $request
+     * @param Stack|null       $stack
+     */
+    private function collectRequestInformation(RequestInterface $request, Stack $stack = null)
```

Comment:
So running this does only have an impact when cache plugin and "without re-validation"?